mods/cnc/maps/shellmap/shellmap.lua

```diff
@@ -1,3 +1,362 @@
+			a.AttackMove(MCVDeploy.Location)
+		end},
+		{{"bike", "bike"},function(a)
+			a.AttackMove(HandBuild.Location+CVec.New(3,5))
+			a.AttackMove(MCVDeploy.Location)
+		 end},
+		{{"bggy", "bggy", "bggy"},function(a)
+			a.AttackMove(HandBuild.Location+CVec.New(3,5))
+			a.AttackMove(MCVDeploy.Location)
+		 end}
+	}
+OutpostInfantry =
+	{
+		{{"e1", "e1", "e3"}, function(a) a.AttackMove(MCVDeploy.Location) end },
+		{{"e1", "e1", "e1", "e3", "e3"}, function(a) a.AttackMove(MCVDeploy.Location) end },
+		{{"e4", "e4"}, function(a) a.AttackMove(MCVDeploy.Location) end },
+		{{"e1", "e1", "e1", "e1", "e3", "e3", "e4", "e4" }, function(a) a.AttackMove(MCVDeploy.Location) end }
+	}
+
+LSTREntryPath = {LSTRStart.Location, LSTRMove.Location, LSTRLand.Location}
+LSTMEntryPath = {LSTMStart.Location, LSTMMove.Location, LSTMLand.Location}
+LSTLEntryPath = {LSTLStart.Location, LSTLMove.Location, LSTLLand.Location}
+
+LSTRExitPath = {CPos.New(LSTRMove.Location.X,LSTLStart.Location.Y), CPos.New(8,LSTLStart.Location.Y)}
+LSTMExitPath = {LSTMMove.Location, CPos.New(8,LSTMStart.Location.Y)}
+LSTLExitPath = {CPos.New(LSTLMove.Location.X,LSTRStart.Location.Y), CPos.New(8,LSTRStart.Location.Y)}
+
+--Repeated Functions START
+UnloadLST = function(lst,cargo)
+	Utils.Do(cargo, function()
+		local a = lst.UnloadPassenger()
+		a.Teleport(lst.Location)
+		a.IsInWorld = true
+		a.Move(lst.Location - CVec.New(0, 1))
+	end)
+end
+
+PlayMusic = function()
+	Media.PlayMusic("map1", PlayMusic)
+end
+
+StayInLocation = function(a,loc,dist)
+	if not a.IsDead then
+		local vector = CVec.New(a.Location.X-loc.X,a.Location.Y-loc.Y)
+		if math.abs(vector.X) > dist or math.abs(vector.Y) > dist then
+			a.Stop()
+			a.AttackMove(loc)
+			Trigger.AfterDelay(250, function() StayInLocation(a,loc,dist) end)
+		end
+	end
+end
+MakeNod = function(fact,teams)
+	nod.Cash = nod.Cash + 3000
+	if not fact.IsDead then
+		local teamtype = Utils.Random(teams)
+		fact.Build(teamtype[1], function(team)
+			Utils.Do(team, teamtype[2])
+			MakeNod(fact,teams)
+		end)
+	end
+end
+MakeGDI = function(fact,teamtype)
+	gdi.Cash = gdi.Cash + 3000
+	fact.Build(teamtype, function(team)
+		Utils.Do(team, function(a)
+			a.Stance = "Defend"
+			StayInLocation(a,MCVDeploy.Location-CVec.New(3,7),1)
+		end)
+		Trigger.OnAllKilled(team,function()
+			if not fact.IsDead then
+				MakeGDI(fact,teamtype)
+			end
+		end)
+	end)
+end
+MakeUnitInvincible = function(unit)
+	if not unit.HasProperty("AcceptsUpgrade") then
+		unit.Destroy()
+	end
+	if unit.HasProperty("AcceptsUpgrade") and unit.AcceptsUpgrade("unkillable") then
+		unit.GrantUpgrade("unkillable")
+		if unit.Stance ~= nil then
+			unit.Stance = "Defend"
+		end
+	end
+end
+--Repeated Functions END
+
+--Event Sequence Functions START
+SendCommandos = function()
+	local commandos = Reinforcements.ReinforceWithTransport(gdi, "tran", CommandoTeam, 
+	{ChinEntry.Location, ChinUnload.Location}, {ChinExit.Location})[2]
+	
+	Trigger.OnIdle(commandos[1], function()
```

Comment:
Removing the OnIdle caused the commandos to do absolutely nothing. Uising an OnAddedToWorld might work equally well but OnIdle means they are given something to do if they somehow get lost.